frontend/src/pages/question/panels/NodePanel.jsx

```diff
@@ -151,22 +150,77 @@ export default function NodePanel({ panelStore }) {
     // so that they show up in the list in the
     // right order
     const combinedConcepts = conceptsFormatted.map(
-      (c, i) => [c, conceptsSetified[i]]).flat();
+      (c, i) => [c, conceptsSetified[i]],
+    ).flat();
 
     node.setConcepts(combinedConcepts);
   }
   // When node panel mounts get concepts
   useEffect(() => { updateConceptList(); }, [biolink]);
 
   async function fetchCuries(newSearchTerm) {
-    // Get and update list of curies
-    const response = await API.ranker.entityLookup(newSearchTerm);
-    if (response.status === 'error' || !_.isArray(response)) {
+    // Get list of curies that match this search term
+    const response = await API.nameResolver.entityLookup(newSearchTerm, 1000);
+    if (response.status === 'error') {
       displayAlert('error',
-        'Failed to contact server to search curies. You will still be able to select generic types. Please try again later');
-    } else {
-      node.updateCuries(response);
+        'Failed to contact name resolver to search curies. You will still be able to select generic types. Please try again later');
+      node.setLoading(false);
+      return;
+    }
+    const curies = Object.keys(response);
+
+    // Pass curies to nodeNormalizer to get type information and
+    // a better curie identifier
+    //
+    // Right now curies are in the URL so we need to split it so we don't
+    // hit maximum URL length of the server
+
+    // Based on experimentation, the max URL length is around 7000 characters
+    // Higher values are faster, so we want to set this as high as possible
+    const maxUrlLength = 7000;
+
+    // Iterate over each curie chunking based on the length that the URL will be
+    const curiesChunked = [[]];
+    curies.forEach((c) => {
+      const existingArray = curiesChunked[curiesChunked.length - 1];
+      const existingLength = existingArray.join('&curie=').length;
+      const newLength = existingLength + c.length + '&curie='.length;
+      if (newLength < maxUrlLength) {
+        existingArray.push(c);
+      } else {
+        curiesChunked.push([c]);
+      }
+    });
+
+    const normalizationAPICallResponses = await Promise.all(
+      curiesChunked.map((cs) => API.nodeNormalization.getNormalizedNodes(cs)),
+    );
```

Comment:
This all looks reasonable. But silly. We should open an issue or something on the Normalizer repo and see if they can accept a POST with body so we don't have to do this.

Reply:
The fix is already in the works, but let's go ahead with a workaround in case they're slow.